### PR TITLE
LibJS: Implement SetValueInBuffer closer to spec

### PR DESCRIPTION
**Pull request by shannonbooth**
No functional changes intended, mostly just adding spec comments, along
with some asserts.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/ArrayBuffer.h [hunk 1]
```diff
@@ -43,6 +43,8 @@ class ArrayBuffer : public Object {
     void set_detach_key(Value detach_key) { m_detach_key = detach_key; }
 
     void detach_buffer() { m_buffer = Empty {}; }
+
+    // 25.1.2.2 IsDetachedBuffer ( arrayBuffer ), https://tc39.es/ecma262/#sec-isdetachedbuffer
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ArrayBuffer.h:49]: This also needs inline comments for step 1 and 2.

File: Userland/Libraries/LibJS/Runtime/ArrayBuffer.h [hunk 2]
```diff
@@ -213,11 +213,43 @@ void ArrayBuffer::set_value(size_t byte_index, Value value, [[maybe_unused]] boo
...
+
+    // 6. If isLittleEndian is not present, set isLittleEndian to the value of the [[LittleEndian]] field of the surrounding agent's Agent Record.
+    //    NOTE: Done by default parameter declaration of this function.
+
+    // 7. Let rawBytes be NumericToRawBytes(type, value, isLittleEndian).
     auto raw_bytes = numeric_to_raw_bytes<T>(vm, value, is_little_endian);
 
-    // FIXME: Check for shared buffer
+    // FIXME 8. If IsSharedArrayBuffer(arrayBuffer) is true, then
+    {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ArrayBuffer.h:240]: `if (false) {`

File: Userland/Libraries/LibJS/Runtime/ArrayBuffer.h [hunk 3]
```diff
@@ -213,11 +213,43 @@ void ArrayBuffer::set_value(size_t byte_index, Value value, [[maybe_unused]] boo
...
+        // FIXME: a. Let execution be the [[CandidateExecution]] field of the surrounding agent's Agent Record.
+        // FIXME: b. Let eventsRecord be the Agent Events Record of execution.[[EventsRecords]] whose [[AgentSignifier]] is AgentSignifier().
+        // FIXME: c. If isTypedArray is true and IsNoTearConfiguration(type, order) is true, let noTear be true; otherwise let noTear be false.
+        // FIXME: d. Append WriteSharedMemory { [[Order]]: order, [[NoTear]]: noTear, [[Block]]: block, [[ByteIndex]]: byteIndex, [[ElementSize]]: elementSize, [[Payload]]: rawBytes } to eventsRecord.[[EventList]].
+    }
+    // 9. Else,
+    {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ArrayBuffer.h:247]: `else {`

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibJS/Runtime/ArrayBuffer.h (hunk 2 above)
Comment (shannonbooth) [reply Userland/Libraries/LibJS/Runtime/ArrayBuffer.h:240]: Ah sure! I wasn't sure what style LibJS would do for that, will use that :)

File: Userland/Libraries/LibJS/Runtime/ArrayBuffer.h (hunk 1 above)
Comment (shannonbooth) [reply Userland/Libraries/LibJS/Runtime/ArrayBuffer.h:49]: Wil do!

**Pull request merged**